modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_slack_osqp_interface.cc

```diff
@@ -0,0 +1,475 @@
+  w_ev_ = ego_(1, 0) + ego_(3, 0);
+  l_ev_ = ego_(0, 0) + ego_(2, 0);
+  g_ = {l_ev_ / 2, w_ev_ / 2, l_ev_ / 2, w_ev_ / 2};
+  offset_ = (ego_(0, 0) + ego_(2, 0)) / 2 - ego_(2, 0);
+  obstacles_edges_sum_ = obstacles_edges_num_.sum();
+
+  l_start_index_ = 0;
+  n_start_index_ = l_start_index_ + obstacles_edges_sum_ * (horizon_ + 1);
+  s_start_index_ = n_start_index_ + 4 * obstacles_num_ * (horizon_ + 1);
+
+  l_warm_up_ = Eigen::MatrixXd::Zero(obstacles_edges_sum_, horizon_ + 1);
+  n_warm_up_ = Eigen::MatrixXd::Zero(4 * obstacles_num_, horizon_ + 1);
+  slacks_ = Eigen::MatrixXd::Zero(obstacles_edges_sum_, horizon_ + 1);
+
+  // get_nlp_info
+  lambda_horizon_ = obstacles_edges_sum_ * (horizon_ + 1);
+  miu_horizon_ = obstacles_num_ * 4 * (horizon_ + 1);
+  slack_horizon_ = obstacles_num_ * (horizon_ + 1);
+
+  // number of variables
+  num_of_variables_ = lambda_horizon_ + miu_horizon_ + slack_horizon_;
+  // number of constraints
+  num_of_constraints_ = 3 * obstacles_num_ * (horizon_ + 1) + num_of_variables_;
+
+  min_safety_distance_ =
+      planner_open_space_config.dual_variable_warm_start_config()
+          .min_safety_distance();
+  check_mode_ =
+      planner_open_space_config.dual_variable_warm_start_config().debug_osqp();
+  beta_ = planner_open_space_config.dual_variable_warm_start_config().beta();
+}
+
+void DualVariableWarmStartSlackOSQPInterface::printMatrix(
+    const int r, const int c, const std::vector<c_float>& P_data,
+    const std::vector<c_int>& P_indices,
+    const std::vector<c_int>& P_indptr) {
+  Eigen::MatrixXf tmp = Eigen::MatrixXf::Zero(r, c);
+
+  for (size_t i = 0; i < P_indptr.size() - 1; ++i) {
+    if (P_indptr[i] < 0 || P_indptr[i] >= static_cast<int>(P_indices.size())) {
+      continue;
+    }
+
+    for (auto idx = P_indptr[i]; idx < P_indptr[i + 1]; ++idx) {
+      int tmp_c = static_cast<int>(i);
+      int tmp_r = static_cast<int>(P_indices[idx]);
+      tmp(tmp_r, tmp_c) = static_cast<float>(P_data[idx]);
+    }
+  }
+
+  AINFO << "row number: " << r;
+  AINFO << "col number: " << c;
+  for (int i = 0; i < r; ++i) {
+    AINFO << "row number: " << i;
+    AINFO << tmp.row(i);
+  }
+}
+
+void DualVariableWarmStartSlackOSQPInterface::assembleA(
+    const int r, const int c, const std::vector<c_float>& P_data,
+    const std::vector<c_int>& P_indices, const std::vector<c_int>& P_indptr) {
+  constraint_A_ = Eigen::MatrixXf::Zero(r, c);
+
+  for (size_t i = 0; i < P_indptr.size() - 1; ++i) {
+    if (P_indptr[i] < 0 || P_indptr[i] >= static_cast<int>(P_indices.size())) {
+      continue;
+    }
+
+    for (auto idx = P_indptr[i]; idx < P_indptr[i + 1]; ++idx) {
+      int tmp_c = static_cast<int>(i);
+      int tmp_r = static_cast<int>(P_indices[idx]);
+      constraint_A_(tmp_r, tmp_c) = static_cast<float>(P_data[idx]);
+    }
+  }
+}
+
+bool DualVariableWarmStartSlackOSQPInterface::optimize() {
+  int kNumParam = num_of_variables_;
+  int kNumConst = num_of_constraints_;
```

Comment:
Only const variables follow the `kXxx` name pattern.

Reply:
> Only const variables follow the `kXxx` name pattern.

Fixed